### ISSUE TITLE
Ray[tune] gets stuck after "leaked semaphore" warning 

### ISSUE THREAD

**Issue by Fritskee** [ultralytics/ultralytics#18252]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Hyperparameter Tuning

### Bug

I am trying to conduct a hyperparameter tuning of our yolo-v8 obb model. For this I am using ray tune and it's `ASHAScheduler` that is implemented in ultralytics.

By default the scheduler tries to maximize the mAP50-95(B) metric. However, I am interested in a different metric. This however is where the problems arise. 

Inside `ultralytics/utils/tuner.py` the ASHAScheduler uses `metric=TASK2METRIC[task]` to set the metric. As you can see, the TASK2METRIC function returns a string. In the case for obb, this string is 'metrics/mAP50-95(B)'
<img width="189" alt="Scherm­afbeelding 2024-12-16 om 08 34 07" src="https://github.com/user-attachments/assets/da92eb58-a25e-4118-8a11-929a4ff6d882" />

<img width="413" alt="image (14)" src="https://github.com/user-attachments/assets/e3fc0e7e-d185-440d-98d0-e0a81073b2a5" />

When I try to change this value however (both in the TASK2METRIC) function, or by directly changing it inside the codebase, the code stops running after a leaked semaphore warning. The weird thing is **this even happens when I just copy the same value returned by TASK2METRIC**. Meaning, the moment I don't call the function, but pass in even the same string, the issue arises. 
<img width="544" alt="image (15)" src="https://github.com/user-attachments/assets/2f7696e9-8633-4064-bf50-5dfec5f0c886" />
I have also tried to use single quotes instead of double quotes, but as expected this also wasn't the issue. 

Blow I show you where the program stalls. Note that this is only when I don't use the TASK2METRIC function, or when I change the value inside the TASK2METRIC function to a different value than "metrics/mAP50-95(B)" for obb. 
<img width="1385" alt="Scherm­afbeelding 2024-12-16 om 08 46 14" src="https://github.com/user-attachments/assets/509b16b5-c505-4108-9210-4a3b3c0bbba6" />

### Environment

Ultralytics 8.3.44 🚀 Python-3.8.8 torch-2.0.1 CPU (Apple M2 Pro)
Setup complete ✅ (10 CPUs, 16.0 GB RAM, 426.1/460.4 GB disk)

OS                  macOS-14.6.1-arm64-arm-64bit
Environment         Darwin
Python              3.8.8
Install             pip
RAM                 16.00 GB
Disk                426.1/460.4 GB
CPU                 Apple M2 Pro
CPU count           10
GPU                 None
GPU count           None
CUDA                None

numpy               ✅ 1.24.3>=1.23.0
numpy               ✅ 1.24.3<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.7.1>=3.3.0
opencv-python       ✅ 4.10.0.82>=4.6.0
pillow              ✅ 9.5.0>=7.1.2
pyyaml              ✅ 6.0>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.9.1>=1.4.1
torch               ✅ 2.0.1>=1.8.0
torch               ✅ 2.0.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.15.2>=0.9.0
tqdm                ✅ 4.65.0>=4.64.0
psutil              ✅ 5.9.5
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.12.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

Version of raytune is: ray 2.10.0

### Minimal Reproducible Example

import os 
from ray import tune
from ultralytics import YOLO

model = YOLO("yolov8n-obb.pt")

result_grid = model.tune(data="dota8.yaml", use_ray=True, epochs=2, grace_period=2, iterations=2)

### Additional

Is only happens when I don't use the TASK2METRIC function, or when I change the value inside the TASK2METRIC function to a different value than "metrics/mAP50-95(B)" for obb. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Fritskee, thank you for your detailed report and for your interest in Ultralytics 🚀!  

We recommend starting with the [documentation](https://docs.ultralytics.com) for general usage and hyperparameter tuning examples. It looks like you are working on custom modifications, specifically involving the Ray Tune integration. Let me guide you further:  

If this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate the issue effectively. You've already included one, which is fantastic—thank you for that! Having this MRE will be instrumental in debugging the potential cause.  

Additionally, since you are testing with custom modifications such as changes to `TASK2METRIC`, which leads to a semaphore warning, ensuring you're operating in a fresh and correctly-configured environment is crucial. Please confirm that your installed `ultralytics` package and dependencies are fully updated. You can do this by running the following command:  

```bash
pip install -U ultralytics
```

You’ll also want to verify that Ray Tune (and its specific version features) is compatible with the changes you are making. If reverting the modifications resolves the issue, then it might indicate specific constraints with how `TASK2METRIC` is interacting with the scheduler or metric handling.

## Environments  

YOLO may be run in any of the following environments with verified dependencies (including [Python](https://www.python.org/) >=3.8 and [PyTorch](https://pytorch.org/) >=1.8 pre-installed):  
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>  
- **Google Cloud** Deep Learning VM: [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)  
- **Amazon AWS** Deep Learning AMI: [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)  
- **Docker Image**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a> [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)  

## Community  

We're here to help! If you’d like additional input or a second opinion from the community or developers:  
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions  
- Visit the [Discourse forum](https://community.ultralytics.com) for in-depth discussions  
- Participate on our [Subreddit](https://reddit.com/r/Ultralytics)  

⚡ **This is an automated response**—an Ultralytics engineer will take a closer look at your issue shortly and provide further support. Thank you again for your patience and for helping improve Ultralytics! 🚀

**Comment by Y-T-G** [comment]
Are you modifying the installed package?

**Comment by Fritskee** [comment]
I'm running ultralytics as project and modifying inside the project

**Comment by Y-T-G** [comment]
You would need to install it by running `pip install -e .` (dot at the end) inside the cloned folder. Otherwise raytune would be using your pip installed package, not the one that you modified.

**Comment by Y-T-G** [comment]
Does `yolo checks` show install type as `git`?

**Comment by Fritskee** [comment]
```
Ultralytics 8.3.3 🚀 Python-3.11.11 torch-2.5.1 CPU (Apple M2 Pro)
Setup complete ✅ (10 CPUs, 16.0 GB RAM, 433.6/460.4 GB disk)

OS                  macOS-14.6.1-arm64-arm-64bit
Environment         Darwin
Python              3.11.11
Install             git
RAM                 16.00 GB
CPU                 Apple M2 Pro
CUDA                None

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.0.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0
```

@Y-T-G yes it does
